flake.nix

```diff
@@ -0,0 +1,164 @@
+      ;
+      inherit (overlay-utils.lib)
+        createOverlays
+      ;
+      inherit (source-utils.lib { inherit (nixpkgs) lib; })
+        filterSource
+      ;
+      inherit (version-utils.lib)
+        getUnstableVersion
+      ;
+
+      supportedSystems = defaultSystems;
+      commonArgs = {
+        version = getUnstableVersion self.lastModifiedDate;
+        homepage = "https://www.skolfederation.se/egil-scimclient-esc/";
+        downloadPage = "https://github.com/Sambruk/EgilSCIM/releases";
+        changelog = "https://raw.githubusercontent.com/Sambruk/EgilSCIM/master/CHANGELOG.md";
+        maintainers = [
+          {
+            name = "ilkecan bozdogan";
+            email = "ilkecan@protonmail.com";
+            github = "ilkecan";
+            githubId = "40234257";
+          }
+        ];
+        platforms = supportedSystems;
+      };
+
+      derivations = {
+        egil-scim-client = import ./nix/egil-scim-client.nix commonArgs;
+        egil-scim-client-debug = import ./nix/egil-scim-client.nix (commonArgs // { isDebugBuild = true; });
+
+        egil-plugins-echo = import ./nix/egil-plugins-echo.nix commonArgs;
+
+        egil-test-server = import ./nix/egil-test-server.nix commonArgs;
+        egil-test-suite = import ./nix/egil-test-suite.nix commonArgs;
+
+        egil-tools = import ./nix/egil-tools/all.nix (commonArgs // { inherit egilToolPackageNames; });
+        egil-tools-fetch_metadata = import ./nix/egil-tools/fetch_metadata.nix commonArgs;
+        egil-tools-list_metadata = import ./nix/egil-tools/list_metadata.nix commonArgs;
+        egil-tools-public_key_pin = import ./nix/egil-tools/public_key_pin.nix commonArgs;
+      };
+
+      packageNames = attrNames derivations;
+      egilToolPackageNames = filter (hasPrefix "egil-tools-") packageNames;
+      debugPackageNames = filter (hasSuffix "-debug") packageNames;
+      nonDebugPackageNames = subtractLists debugPackageNames packageNames;
+      appPackageNames = subtractLists [ "egil-tools" "egil-plugins-echo" ] packageNames;
+    in
+    {
+      overlays = createOverlays derivations { inherit filterSource; };
+      overlay = self.overlays.egil-scim-client;
+    } // eachSystem supportedSystems (system:
+      let
+        inherit (pkgs.stdenv) isLinux;
+
+        packageNamesToSkip = optionals (system != "x86_64-linux") [ "egil-test-suite" ];
+        packageNamesToKeep = subtractLists packageNamesToSkip packageNames;
+
+        overlays = getAttrs packageNamesToKeep self.overlays;
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = attrValues overlays;
+        };
+      in
+      rec {
+        checks = packages;
+
+        packages = getAttrs packageNamesToKeep pkgs;
+        defaultPackage = packages.egil-scim-client;
+
+        apps = mapAttrs (name: drv: mkApp { drv = getBin drv; }) (getAttrs (intersectLists packageNamesToKeep appPackageNames) packages);
+        defaultApp = apps.egil-scim-client;
```

Comment:
This one is probably not necessary as long as `packages` are configured correctly.
`nix run` checks `apps`, if app is not found then checks `packages` and then `legacyPackages`

Reply:
Yes, but `nix run` tries the package name as the executable name, which is not true most of the. I don't know a way to specify the executable name from the derivation with vanilla nix so I passed it with 
```
  passthru = {
    inherit exePath;
  };
```
which is checked and used by `mkApp` provided by `flake-utils`

Reply:
`meta.mainProgram` is what you are looking for

> If  installable  evaluates to a derivation, it will try to execute the program <out>/bin/<name>,
       where out is the primary output store path of the derivation and name  is  the  meta.mainProgram
       attribute  of  the derivation if it exists, and otherwise the name part of the value of the name
       attribute of the derivation (e.g. if name is set to hello-1.10, it will run $out/bin/hello)

Reply:
Resolved with 18428fd13de46ba925cca302252c10ad5e5aaede